src/views/catalog/customize/components/CustomizeForms/useCustomizeFormSubmit.ts

```diff
@@ -0,0 +1,56 @@
+import * as React from 'react';
+import { useHistory, useParams } from 'react-router-dom';
+
+import { V1Template } from '@kubevirt-ui/kubevirt-api/console';
+import { getAnnotation } from '@kubevirt-utils/resources/shared';
+import { ANNOTATIONS } from '@kubevirt-utils/resources/template';
+import { getTemplateVirtualMachineObject } from '@kubevirt-utils/resources/template/utils/selectors';
+
+import { useWizardVMContext } from '../../../utils/WizardVMContext';
+import { processTemplate } from '../../utils';
+
+type useCustomizeFormSubmitType = [
+  onSubmit: (event: any) => Promise<void>,
+  loaded: boolean,
+  error: any,
+];
+
+export const useCustomizeFormSubmit = (template: V1Template): useCustomizeFormSubmitType => {
```

Comment:
why do you need a use for submit ?

it should be a synchronous operation

Reply:
just to have a clean code inside the component and why not maybe reuse some logic for the other forms.